_posts/ref/2022-01-18-wire-v1.md

```diff
@@ -0,0 +1,208 @@
+A valid object in this format can be obtained by navigating to the **Wires** section of ESF web UI, clicking the **Download** button and selecting the JSON format download.
+
+Accessing the REST APIs requires to use an identity with the `rest.configuration` permission assigned.
+## Request definitions
+#### GET/graph
+  * **REST API path** : /services/wire/v1/graph
+  * **description** : Returns the current Wire Graph Configuration as a [ComponentConfigurationList]({{ site.baseurl }}{% link _posts/ref/2021-12-13-conf-v2.md %}#componentconfigurationlist) object. The received configuration will include the [WireGraphService]({{ site.baseurl }}{% link _posts/ref/2022-01-18-wire-graph-service.md %}) configuration containing the graph layout and the configuration of the components currently referenced by the Wire Graph.
+  * **responses** :
+    * **200**
+      * **description** : The current wire graph configuration.
+      * **response body** :
+        * [ComponentConfigurationList]({{ site.baseurl }}{% link _posts/ref/2021-12-13-conf-v2.md %}#componentconfigurationlist) 
+    * **500**
+      * **description** : An unexpected internal error occurred.
+      * **response body** :
+        * [GenericFailureReport]({{ site.baseurl }}{% link _posts/ref/2021-12-13-conf-v2.md %}#genericfailurereport)
+
+#### PUT/graph
+  * **REST API path** : /services/wire/v1/graph
+  * **description** : Updates the current Wire Graph.
+  * **request body** :
+    * [ComponentConfigurationList]({{ site.baseurl }}{% link _posts/ref/2021-12-13-conf-v2.md %}#componentconfigurationlist) 
+  * **responses** :
+    * **200**
+      * **description** : The current Wire Graph has been updated.
+    * **400**
+      * **description** : The request body is not valid JSON or it contains invalid parameters.
+      * **response body** :
+        * [GenericFailureReport]({{ site.baseurl }}{% link _posts/ref/2021-12-13-conf-v2.md %}#genericfailurereport)
+    * **500**
+      * **description** : An unexpected internal error occurred.
+      * **response body** :
+        * [GenericFailureReport]({{ site.baseurl }}{% link _posts/ref/2021-12-13-conf-v2.md %}#genericfailurereport)
+
+This request will replace the current graph topology with the received one.
+
+* Wire Components and Wires that are specified in the request and do not exist on the current graph on the device will be created.
+* Wire Components and Wires that exist in the current graph but are not included in the request will be deleted.
+* Wire Component configurations and rendering properties of existing Wire Components will be updated.
+
+The received configuration must satisfy the following requirements, if any of the requirements is not met, the operation will fail and no changes will be applied to the system:
```

Comment:
> The received configuration must satisfy the following requirements, if any of the requirements is not met, the operation will fail and no changes will be applied to the system:

The received configuration must satisfy the following requirements. If any of the requirements is not met, the operation will fail and no changes will be applied to the system: